libc/sock/socket-nt.c

```diff
@@ -19,12 +19,19 @@
 #include "libc/calls/internal.h"
 #include "libc/mem/mem.h"
 #include "libc/nt/winsock.h"
+#include "libc/nt/iphlpapi.h"
 #include "libc/sock/internal.h"
 #include "libc/sock/yoink.inc"
 #include "libc/sysv/consts/fio.h"
 #include "libc/sysv/consts/o.h"
 #include "libc/sysv/consts/sock.h"
 
+/* ioctl(SIOCGIFCONFIG) for Windows need to access the following functions through 
+ * weak reference. This ensure those symbols are not stripped during final link
+ */
+STATIC_YOINK("GetAdaptersAddresses");
```

Comment:
The optimal place for these yoinks might actually be in the `__IOCTL_DISPATCH()` macro, similar to the hack we're using for `__SIGACTION_YOINK()`. It also technically shouldn't be a problem for code in `LIBC_CALLS` to depend on `LIBC_NT_FOO` or `LIBC_STR` since those packages are higher up in the topological order of things and would not create a cyclic dependency.